### ISSUE TITLE
Outlier detection / Destination rules not updated (CDS reporting "Not Sent")

### ISSUE THREAD

**Issue by arnecls** [istio/istio#19902]
**Bug description**

When having a destination rule with outlier detection enabled inside your request path, outlier detection will be triggered during a pilot restart. 

Note: This issue was first noticed with istio 1.3.x. It does not appear with istio 1.2.x.
If we leave outlier detection on, we also see these issues happening after some time without pilot restarts, but we cannot really pin it down what causes them to happen.

**Expected behavior**

Outlier detection should not be affected by pilot restarts.

**Steps to reproduce the bug**

I have no simple rule for reproduction, yet. This is what we see on our production clusters:
- We have a request path of ~5 grpc-services until the service using outlier detection is accessed.
- The service target by outlier detection has a virtual service, setting retries.attempts to 0
- The service target by outlier detection has two ports open, one called 'grpc', the other called 'http'. The 'grpc' port is used for requests.
- The communication to the service target by outlier detection happens within the same namespace
- Subsets are not used
- Pilot is running with 2 replicas in 2 different zones (see values.yaml below)

Destination rule used (targeting the service mentioned above)
```
spec:
  trafficPolicy:
    outlierDetection:
      baseEjectionTime: 3s
      consecutiveErrors: 2
      interval: 500ms
      maxEjectionPercent: 35
``` 

Immediately after adding this, we restart pilot using
```
kubectl patch deployment -n istio-system istio-pilot -p "{\"spec\":{\"template\":{\"metadata\":{\"labels\":{\"date\":\"$(date +'%s')\"}}}}}"
```

Then we run a couple of tests against our services, resulting in an increased number of errors (requests are failing). The errors go down and finally disappear as the new pilot instance become ready.
If we remove the outlier detection from the destination rule and restart pilot, all tests are running fine, regardless of restarting pilot.

```
istioctl version --remote
client version: 1.4.2
control plane version: 1.4.2
data plane version: 1.4.2 (132 proxies)
```

```
kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.7", GitCommit:"8fca2ec50a6133511b771a11559e24191b1aa2b4", GitTreeState:"clean", BuildDate:"2019-09-18T14:47:22Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.8-gke.17", GitCommit:"188432a69210ca32cafded81b4dd1c063720cac0", GitTreeState:"clean", BuildDate:"2019-11-13T20:47:11Z", GoVersion:"go1.12.11b4", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**

Rendering of helm charts using helm2, then using `kubectl apply` on the rendered manifests.
Helm values file [values.yaml.zip](https://github.com/istio/istio/files/4019056/values.yaml.zip)

**Comment by arnecls** [comment]
After playing around a bit, we found out that destination rules are not properly synced in our mesh.
I.e. if I apply a destination rule with outlier detection, and do an `istioctl proxy-config cluster` on the cluster I just configured, there is no change visible. Even after a minute or two.

What we also see is that `istioctl proxy-status` has a lot of workloads marked as `Not Sent` but _only_ in the CDS column.
We checked pilot error logs, but the only thing we see is a `pilot_conflict_outbound_listener_tcp_over_current_tcp` on port 443 (ip 0.0.0.0).
However, this apparently does not have any effect, as we can still reach everything on port 443.
As of _why_ there is a collision, we're not sure, yet, but we assume it's because of one wildcard dns service entry that we have in use (*.googleapis.com). Also this error should only affect listeners, not clusters.

So the question is left, why pilot can not update CDS for over 50% of the sidecars, while all other parts are working fine ...?

**Comment by valer-cara** [comment]
I'm having a very similar issue with CDS not sent after updating a DestinationRule. I'm not using outlier detection though. It's really hard to track down.

Tried restarting pilot but i found no way to force it to sync CDS. The rest of the xDS is sync'ed. Haven't spotted anything in the logs so far, except for the fact that `istio-pilot/discovery` doesn't print the CDS updated lines any longer.

I'm running in a multi-cluster setup with replicated control planes. I initially thought it might have smth to do with merging of DestinationRules, since there's a main `*.global` rule and I had added another `prometheus.monitoring.global`, but couldn't confirm.. Any hints much appreciated, it's killing me :<

PS: I'm running 1.4-dev; installed via the operator.
